### PR TITLE
PWN-7174 - New swap. Token list A logic

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7174

## Description of Work
- fetching current token a, fetching list of tokens 
- sorting user tokens by USDC, USDT, or usd amount

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateManager.kt [hunk 1]
```diff
@@ -21,6 +23,10 @@ class SwapStateManager(
 
     fun observe(): StateFlow<SwapState> = state
 
+    suspend fun <T> getStateValue(getter: (state: SwapState) -> T): T {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateManager.kt:26]: is it okay to get token a and b from state? 
@NikolaiEmelianov

Comment (NikolaiEmelianov) [reply app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateManager.kt:41]: better to listen the Flow. 
If tokenA change you reinstall screen state
If catch some wrong state then you close screen or something
But in this local feature case Ok. I will disable click for wrong state

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/mapper/SwapWidgetMapper.kt [hunk 2]
```diff
@@ -97,7 +97,12 @@ class SwapWidgetMapper {
 
     private fun tokenName(token: SwapTokenModel): TextViewCellModel.Raw =
         TextViewCellModel.Raw(
-            TextContainer(token.tokenName)
```
Comment (NikolaiEmelianov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/mapper/SwapWidgetMapper.kt:100]: no need

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/mapper/SwapWidgetMapper.kt [hunk 3]
```diff
@@ -123,7 +128,7 @@ class SwapWidgetMapper {
         )
 
     private fun tokenAmount(token: SwapTokenModel.UserToken): String =
-        token.details.total.formatToken(token.details.decimals)
+        token.tokenAmount.formatToken(token.details.decimals)
```
Comment (NikolaiEmelianov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/mapper/SwapWidgetMapper.kt:131]: thanks

File: app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateManager.kt (hunk 1 above)
Comment (NikolaiEmelianov) [reply app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateManager.kt:41]: click that open this screen

File: app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TokenViewHolder.kt [hunk 4]
```diff
@@ -82,7 +82,7 @@ class TokenViewHolder(
     fun bindBalance(item: HomeElementItem.Shown) {
         renderItem = item
         binding.valueTextView withTextOrGone token.getFormattedUsdTotal()
-        binding.totalTextView.text = token.getTotal(includeSymbol = true)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TokenViewHolder.kt:85]: getTotal == getFormattedTotal
same logic, DRY used

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/adapter/SwapTokensAdapter.kt [hunk 5]
```diff
@@ -30,7 +33,12 @@ class SwapTokensAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
             SectionHeaderViewHolder(parent.inflateViewBinding(attachToRoot = false))
         }
         FinanceBlockViewHolder.DEFAULT_VIEW_TYPE -> {
-            FinanceBlockViewHolder(parent.inflateViewBinding(attachToRoot = false))
+            FinanceBlockViewHolder(
+                binding = parent.inflateViewBinding(attachToRoot = false),
+                inflateListener = {
+                    it.setOnClickAction { _, item -> onTokenClicked.invoke(item.payload as SwapTokenModel) }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/adapter/SwapTokensAdapter.kt:39]: payload handler @NikolaiEmelianov

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/interactor/SwapTokensInteractor.kt [hunk 6]
```diff
@@ -1,9 +1,46 @@
...
+    private val homeLocalRepository: HomeLocalRepository,
+    private val swapTokensRepository: JupiterSwapTokensRepository,
+    private val swapRoutesRepository: JupiterSwapRoutesRepository,
+//    private val swapStateManager: SwapStateManager
+) {
+    suspend fun getCurrentTokenA(): SwapTokenModel {
+        return SwapTokenModel.UserToken(
+            homeLocalRepository.getUserTokens().last()
+        )
+//        return swapStateManager.getStateValue { state ->
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/interactor/SwapTokensInteractor.kt:19]: will be removed in next pr

File: core/src/main/java/org/p2p/core/token/Token.kt [hunk 7]
```diff
@@ -78,20 +79,15 @@ sealed class Token constructor(
...
                 visibility == TokenVisibility.DEFAULT
 
-        fun getFormattedUsdTotal(): String? = totalInUsd?.asUsd()
+        fun getFormattedUsdTotal(includeSymbol: Boolean = true): String? {
+            return if(includeSymbol) totalInUsd?.asUsd() else totalInUsd?.formatFiat()
+        }
 
         fun getFormattedTotal(includeSymbol: Boolean = false): String =
             if (includeSymbol) {
-                "${total.formatToken()} $tokenSymbol"
```
Comment (gslevinkov) [inline core/src/main/java/org/p2p/core/token/Token.kt:85]: strange diff, result looks like this
```
        fun getFormattedTotal(includeSymbol: Boolean = false): String =
            if (includeSymbol) {
                "${total.formatToken(decimals)} $tokenSymbol"
            } else {
                total.formatToken(decimals)
            }
```

please verify @kabramovp2p 
we were not using decimals, is this a bug?

File: ui-kit/src/main/kotlin/org/p2p/uikit/organisms/UiKitToolbar.kt [hunk 8]
```diff
@@ -34,6 +34,7 @@ class UiKitToolbar @JvmOverloads constructor(
     ) {
         inflateMenu(menuRes)
 
+
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/organisms/UiKitToolbar.kt:37]:
```suggestion
```

File: ui-kit/src/main/kotlin/org/p2p/uikit/utils/text/TextViewCellModel.kt [hunk 9]
```diff
@@ -115,9 +114,9 @@ fun TextView.bind(model: TextViewCellModel.Raw) {
             backgroundTintList = initialTextStyle.backgroundTint
         }
     setHintTextColor(initialTextStyle.hintTextColors)
-    if (foreground is SkeletonDrawable) {
-        foreground = null
-    }
+//    if (foreground is SkeletonDrawable) {
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/utils/text/TextViewCellModel.kt:117]: revert

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/mapper/SwapWidgetMapper.kt (hunk 2 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/mapper/SwapWidgetMapper.kt:100]: fixed

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/interactor/SwapTokensInteractor.kt (hunk 6 above)
Comment (NikolaiEmelianov) [reply app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/interactor/SwapTokensInteractor.kt:19]: you mean will be uncomment?

**Review by NikolaiEmelianov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/interactor/SwapTokensInteractor.kt (hunk 6 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/interactor/SwapTokensInteractor.kt:19]: yep

**Pull request merged**